<?xml version="1.0" encoding="UTF-8"?>

<launch>
  <arg name="start_half_sitting" default="true"/>
  <arg name="world_name" default="bauzil"/>
  <arg name="world_filename" default="$(find os1_64_bauzil)/worlds/$(arg world_name).world"/>
  <arg name="gzpose" default="-x -4.2 -y 0.0 -z 1.21 -R 0.0 -P 0.0 -Y 3.14" unless="$(arg start_half_sitting)"/>
  <arg name="gui" default="true"/>
  <arg name="debug" default="false"/>
  <arg name="recording" default="false"/>
  <arg name="extra_gazebo_args" default=""/>

  <arg name="viz" default="false"/>

  <arg name="load_model" default="true"/> <!-- AS: should probably be false by default -->
  <arg name="robot"  default="full_v2_sensors"/>          <!-- full, lower_body, foot -->
  <arg name="foot_collision" default="thinbox"/>
  <arg name="enable_leg_passive" default="false"/>
  <arg name="enable_fixed_robot" default="false"/>
  <arg name="default_configuration_type" default="zeros"/>

  <env name="GAZEBO_MODEL_PATH" value="$(find os1_64_bauzil)/models:$(optenv GAZEBO_MODEL_PATH)"/>

  <arg name="left_leg_pose" default="-J leg_left_1_joint 0.0 -J leg_left_2_joint 0.0 -J leg_left_3_joint -0.411354 -J leg_left_4_joint 0.859395 -J leg_left_5_joint -0.448041 -J leg_left_6_joint -0.001708"/>
  <arg name="right_leg_pose" default="-J leg_right_1_joint 0.0 -J leg_right_2_joint 0.0 -J leg_right_3_joint -0.411354 -J leg_right_4_joint 0.859395 -J leg_right_5_joint -0.448041 -J leg_right_6_joint -0.001708"/>
  <arg name="left_arm_pose" default="-J arm_left_1_joint 0.25847 -J arm_left_2_joint 0.173046 -J arm_left_3_joint -0.0002 -J arm_left_4_joint -0.525366 -J arm_left_5_joint 0.0 -J arm_left_6_joint 0.0 -J arm_left_7_joint 0.1"/>
  <arg name="right_arm_pose" default="-J arm_right_1_joint -0.25847 -J arm_right_2_joint -0.173046 -J arm_left_3_joint 0.0002 -J arm_right_4_joint -0.525366 -J arm_right_5_joint 0.0 -J arm_right_6_joint 0.0 -J arm_right_7_joint 0.1"/>
  <arg name="torso_pose" default="-J torso_1_joint 0.0 -J torso_2_joint 0.006761"/>
  <arg name="gzpose" default="-x -4.2 -y 0.0 -z 1.21 -R 0.0 -P 0.0 -Y 3.14 $(arg left_leg_pose) $(arg right_leg_pose) $(arg left_arm_pose) $(arg right_arm_pose) $(arg torso_pose)" if="$(arg start_half_sitting)"/>

  <!-- start up world -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(arg world_filename)"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="recording" value="$(arg recording)"/>
    <arg name="debug" value="$(arg debug)"/>
    <arg name="extra_gazebo_args" default="$(arg extra_gazebo_args)"/>
  </include>

  <group if="$(arg load_model)">
    <include file="$(find talos_data)/robots/upload.launch" >
      <arg name="robot" value="$(arg robot)"/>
      <arg name="foot_collision" default="$(arg foot_collision)"/>
      <arg name="enable_leg_passive" default="$(arg enable_leg_passive)"/>
      <arg name="enable_fixed_robot" default="$(arg enable_fixed_robot)"/>
      <arg name="default_configuration_type" default="$(arg default_configuration_type)"/>
    </include>
  </group>

  <!-- spawn robot in simulation -->
  <!-- <include file="$(find talos_gazebo)/launch/talos_spawn.launch"> -->
  <include file="$(find talos_data)/launch/talos_spawn.launch">
    <arg name="robot"  value="$(arg robot)"/>
    <arg name="gzpose" value="$(arg gzpose)"/>
  </include>

  <!-- default controllers -->
  <!-- <include file="$(find talos_controller_configuration)/launch/talos_default_controllers.launch"/> -->
  <!-- <include file="$(find talos_bringup)/launch/talos_bringup.launch" /> -->
  <include file="$(find talos_data)/launch/talos_bringup.launch" />

  <!-- start rviz for data visualization -->
  <group if="$(arg viz)">
    <node name="rviz_os1_64_bauzil" pkg="rviz" type="rviz" respawn="false" output="screen" args="-d $(find os1_64_bauzil)/rviz/talos.rviz"/>
  </group>

</launch>
